pkg/controllers/reconciler.go

```diff
@@ -167,14 +168,30 @@ func (r *reconciler) createObjects(ctx context.Context,
 	return result, nil
 }
 
-func UpgradeImages(image *string, initimage *string) (upgrade bool) {
-	for _, s := range []*string{image, initimage} {
+func UpgradeImages(image *string, initimage *string, kind string) (upgrade bool) {
+	for i, s := range []*string{image, initimage} {
 		if s == nil {
 			continue
 		}
 
 		if parts := strings.SplitN(*s, ":", 2); len(parts) == 2 && len(parts[0]) > 0 {
 			name, version := parts[0], parts[1]
+
+			var envVarValue string
+
+			if i == 1 {
+				envVarValue = os.Getenv(kind + "_IMAGE")
+			} else {
+				envVarValue = os.Getenv(kind + "_INITIMAGE")
+			}
+
+			if envVarValue != "" && *s != envVarValue {
```

Comment:
these new lines do not seem to depend on `name` or `version` so splitting can be avoided if the envVariables are set?